### PR TITLE
Allow using the `@builder` directive on fields

### PR DESCRIPTION
**Pull request by olivernybroe**
Add field support for builder directive.

- [x] Added or updated tests
- [ ] Documented user facing changes
- [ ] Updated CHANGELOG.md

**Changes**

Adds field definition to builder directive.

**Breaking changes**

### CODE REVIEW

**Comment by olivernybroe** [comment]
@spawnia Not sure if you wanted to make one big pr where we add field definitions to all the directives missing it or if we should do it individually 👍

File: src/Schema/Directives/BuilderDirective.php [hunk 1]
```diff
@@ -35,4 +42,22 @@ public function handleBuilder($builder, $value): object
...
+    /**
+     * Dynamically call a user-defined method to enhance the builder on a field.
+     *
+     * @param  \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder  $builder
+     * @return \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder
+     */
+    public function handleFieldBuilder(object $builder): object
+    {
+        $resolver = $this->getResolverFromArgument('method');
+        $value = $this->directiveArgValue('value');
```
Comment (spawnia) [inline src/Schema/Directives/BuilderDirective.php:55]: `null` might be a valid value to pass, use `directiveHasArg` to check if `value` is defined at all.

**Comment by spawnia** [comment]
One by one is fine with me, thanks!

One other thing: I appreciate the effort and discipline with the conventional commit messages, but I would prefer to stick to the current convention for now. I can see the benefits when paired with tooling that autogenerates changelogs or even releases and strong, written-down conventions that are universally agreed upon, but find them somewhat less readable then a descriptive sentence.

**Comment by olivernybroe** [comment]
@spawnia Alright :) 

I changed it to just a message.

File: src/Schema/Directives/BuilderDirective.php [hunk 2]
```diff
@@ -19,7 +20,13 @@ public static function definition(): string
   If you pass only a class name, the method name defaults to `__invoke`.
   """
   method: String!
-) repeatable on ARGUMENT_DEFINITION | INPUT_FIELD_DEFINITION
+
+  """
+  Pass a value to the builder method.
```
Comment (spawnia) [inline src/Schema/Directives/BuilderDirective.php:25]:
```suggestion
  Pass a value to the method as the second argument after the query builder.
```

**Review by spawnia — COMMENTED** [review]
Implementation and tests look great. Now we just need docs & changelog.

- The description of the directive could be more general now, since it does no longer use an argument necessarily.
- The directive definition must be copied to the api reference docs, and an example usage there would also be nice.

**Comment by olivernybroe** [comment]
@spawnia Added some docs to it 👍

**Pull request merged**